core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-select-existing-file/components/dot-sidebar/dot-sidebar.component.ts

```diff
@@ -1,29 +1,83 @@
+import { faker } from '@faker-js/faker';
+
+import { SlicePipe } from '@angular/common';
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    inject,
+    input,
+    output,
+    signal
+} from '@angular/core';
 
 import { TreeNode } from 'primeng/api';
-import { TreeModule } from 'primeng/tree';
+import { SkeletonModule } from 'primeng/skeleton';
+import { TreeModule, TreeNodeExpandEvent } from 'primeng/tree';
+
+import { TruncatePathPipe } from '../../../../../../pipes/truncate-path.pipe';
 
 @Component({
     selector: 'dot-sidebar',
     standalone: true,
-    imports: [TreeModule],
+    imports: [TreeModule, SlicePipe, TruncatePathPipe, SkeletonModule],
     templateUrl: './dot-sidebar.component.html',
     styleUrls: ['./dot-sidebar.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class DotSideBarComponent {
     /**
-     * An observable that emits an array of TreeNode objects representing folders.
+     * A readonly private field that holds an instance of ChangeDetectorRef.
+     * This is used to detect and respond to changes in the component's data-bound properties.
+     */
+    readonly #cd = inject(ChangeDetectorRef);
+    /**
+     * An observable that emits an array of TreeNode objects representing the folders.
      *
      * @type {Observable<TreeNode[]>}
      * @alias folders
      */
     $folders = input.required<TreeNode[]>({ alias: 'folders' });
     /**
-     * Represents a loading state for the component.
+     * A boolean observable that indicates the loading state.
      *
      * @type {boolean}
-     * @alias loading
      */
     $loading = input.required<boolean>({ alias: 'loading' });
+
+    /**
+     * Signal that generates an array of strings representing percentages.
+     * Each percentage is a random value between 75% and 100%.
+     * The array contains 50 elements.
+     *
+     * @returns {string[]} An array of 50 percentage strings.
+     */
+    $fakeColumns = signal<string[]>(Array.from({ length: 50 }).map((_) => this.getPercentage()));
+
+    /**
+     * Event emitter for when a tree node is expanded.
+     *
+     * This event is triggered when a user expands a node in the tree structure.
+     * It emits an event of type `TreeNodeExpandEvent`.
+     */
+    onNodeExpand = output<TreeNodeExpandEvent>();
+
+    /**
+     * Triggers change detection manually.
+     * This method is used to ensure that the view is updated when the model changes.
+     * It calls the `detectChanges` method on the ChangeDetectorRef instance.
+     */
+    detectChanges() {
+        this.#cd.detectChanges();
+    }
+    /**
+     * Generates a random percentage string between 75% and 100%.
+     *
+     * @returns {string} A string representing a percentage between 75% and 100%.
+     */
+    getPercentage(): string {
+        const number = faker.number.int({ max: 100, min: 75 });
+
+        return `${number}%`;
+    }
```

Comment:
why are we doing this?

Reply:
The purpose is to generate widths between 75% and 100% for the skeleton

![Screenshot 2024-11-06 at 10 15 11 AM](https://github.com/user-attachments/assets/3107a795-c6c5-4696-aae6-8710ef0fa030)

Reply:
Got it. Should we create a service? Looks like something we can reuse 

It can be in another ticket

Reply:
Nice detail and I liked it, but, can be a utility with a plain js `export const randomNumber = (min, max) => Math.floor(Math.random() * (max - min + 1)) + min` not a full library.